Sources/SourceGraph/SourceGraph.swift

```diff
@@ -5,23 +5,23 @@ public final class SourceGraph {
     // Global shared instance to prevent costly deinitialization.
     public static var shared = SourceGraph()
 
-    private(set) var allDeclarations: Set<Declaration> = []
-    private(set) var usedDeclarations: Set<Declaration> = []
-    private(set) var redundantProtocols: [Declaration: (references: Set<Reference>, inherited: Set<Reference>)] = [:]
-    private(set) var rootDeclarations: Set<Declaration> = []
-    private(set) var redundantPublicAccessibility: [Declaration: Set<String>] = [:]
-    private(set) var rootReferences: Set<Reference> = []
-    private(set) var allReferences: Set<Reference> = []
-    private(set) var retainedDeclarations: Set<Declaration> = []
-    private(set) var ignoredDeclarations: Set<Declaration> = []
-    private(set) var assetReferences: Set<AssetReference> = []
-    private(set) var mainAttributedDeclarations: Set<Declaration> = []
-    private(set) var allReferencesByUsr: [String: Set<Reference>] = [:]
-    private(set) var indexedModules: Set<String> = []
-    private(set) var indexedSourceFiles: [SourceFile] = []
-    private(set) var unusedModuleImports: Set<Declaration> = []
-    private(set) var assignOnlyProperties: Set<Declaration> = []
-    private(set) var extensions: [Declaration: Set<Declaration>] = [:]
+    public var allDeclarations: Set<Declaration> = []
+    public var usedDeclarations: Set<Declaration> = []
+    public var redundantProtocols: [Declaration: (references: Set<Reference>, inherited: Set<Reference>)] = [:]
+    public var rootDeclarations: Set<Declaration> = []
+    public var redundantPublicAccessibility: [Declaration: Set<String>] = [:]
+    public var rootReferences: Set<Reference> = []
+    public var allReferences: Set<Reference> = []
+    public var retainedDeclarations: Set<Declaration> = []
+    public var ignoredDeclarations: Set<Declaration> = []
+    public var assetReferences: Set<AssetReference> = []
+    public var mainAttributedDeclarations: Set<Declaration> = []
+    public var allReferencesByUsr: [String: Set<Reference>] = [:]
+    public var indexedModules: Set<String> = []
+    public var indexedSourceFiles: [SourceFile] = []
+    public var unusedModuleImports: Set<Declaration> = []
+    public var assignOnlyProperties: Set<Declaration> = []
+    public var extensions: [Declaration: Set<Declaration>] = [:]
```

Comment:
Not all of these need to be public, and they should all still use `private(set)`.

Reply:
Made them private(set). Now you can find some mark{*} functions internal and some public. They are used in Indexer. With next PR I'm planning to think about modularization of Indexers, and make these functions internal somewhere